r5dev/sdklauncher/basepanel.cpp

```diff
@@ -641,11 +1017,154 @@ void CUIBaseSurface::VirtualItemToClipboard(const std::unique_ptr<MouseEventArgs
+void CUIBaseSurface::ModManagerEnabledToggle(Forms::Control* pSender) {
+	CUIBaseSurface* pSurface = reinterpret_cast<CUIBaseSurface*>(pSender->FindForm());
+
+	List<uint32_t> lSelected = pSurface->m_ModsListView->SelectedIndices();
+
+	if (!lSelected.Count())
+		return;
+
+	modObject object;
+	for (uint32_t i = 0; i < lSelected.Count(); i++) {
+		object = pSurface->m_ModList[lSelected[i]].m_object;
+		pSurface->m_ModList[lSelected[i]];
+	}
+
+	if (pSurface->m_ManagerEnabledToggle->Checked())
+		object.toggled = true;
+	else if (!pSurface->m_ManagerEnabledToggle->Checked())
+		object.toggled = false;
+
+	for (uint32_t i = 0; i < lSelected.Count(); i++) {
+		if (object.toggled) {
+			pSurface->m_ManagerEnabledLabel->SetText("Enabled");
+			pSurface->m_ModList[lSelected[i]].m_object = object;
+			pSurface->m_ModList[lSelected[i]].m_nLevel = m_modStatusLevel::enabledM;
+		}
+		else {
+			pSurface->m_ManagerEnabledLabel->SetText("Disabled");
+			pSurface->m_ModList[lSelected[i]].m_object = object;
+			pSurface->m_ModList[lSelected[i]].m_nLevel = m_modStatusLevel::disabledM;
+		}
+	}
+	
+	pSurface->m_ModsListView->Refresh();
+
+	std::vector<modObject> vObjects = manager.mods;
+	for (int i = 0; i < vObjects.size(); i++) {
+		if (object.appid == vObjects[i].appid)
+			vObjects[i] = object;
+	}
+	manager.mods = vObjects;
+	for (uint32_t i = 0; i < lSelected.Count(); i++)
+		pSurface->m_ModList[lSelected[i]].m_object = object;
+
+	object.updateJson();
+}
+
 //-----------------------------------------------------------------------------
 // Purpose: gets and handles the virtual item
 // Input  : &pEventArgs - 
 //			*pSender - 
 //-----------------------------------------------------------------------------
+void CUIBaseSurface::GetVirtItemMod(const std::unique_ptr<Forms::RetrieveVirtualItemEventArgs>& pEventArgs, Forms::Control* pSender)
+{
+	CUIBaseSurface* pSurface = reinterpret_cast<CUIBaseSurface*>(pSender->FindForm());
+	if (static_cast<int>(pSurface->m_ModList.size()) <= 0)
+		return;
+
+	pEventArgs->Style.ForeColor = Drawing::Color::White;
+	pEventArgs->Style.BackColor = pSender->BackColor();
+	pSurface->m_ModsListView->SetVirtualListSize(static_cast<int32_t>(pSurface->m_ModList.size()));
+
+	static const Drawing::Color cColor[] =
+	{
+		Drawing::Color(92, 236, 89),   // Enabled
+		Drawing::Color(236, 203, 0),   // Disabled
+		Drawing::Color(236, 28, 0),    // Invalid
+	};
+	static const String svLevel[] =
+	{
+		"Enabled",
+		"Disabled",
+		"Invalid",
+	};
+
+	switch (pEventArgs->SubItemIndex)
+	{
+	case 0:
+		pEventArgs->Style.ForeColor = cColor[pSurface->m_ModList[pEventArgs->ItemIndex].m_nLevel];
+		pEventArgs->Text = svLevel[pSurface->m_ModList[pEventArgs->ItemIndex].m_nLevel];
+		break;
+	case 1:
+		std::string text = "";
+		text = pSurface->m_ModList[pEventArgs->ItemIndex].m_object.name + " By ";
+		for (int i = 0; i < pSurface->m_ModList[pEventArgs->ItemIndex].m_object.authors.size(); i++) {
```

Comment:
Cast the results of 'm_object.authors.size()' to int or (preferably) use size_t for 'i' instead of int. 'std::vector::size' returns a 'size_t'.